python/doc/examples/meta_modeling/polynomial_chaos_metamodel/plot_chaos_conditional_expectation.py

```diff
@@ -0,0 +1,669 @@
+    return parametricPCEFunction
+
+
+# %%
+# The next function creates a sparse PCE using least squares.
+
+
+# %%
+def computeSparseLeastSquaresFunctionalChaos(
+    inputTrain,
+    outputTrain,
+    multivariateBasis,
+    basisSize,
+    distribution,
+    sparse=True,
+):
+    """
+    Create a sparse polynomial chaos based on least squares.
+
+    * Uses the enumerate rule in multivariateBasis.
+    * Uses the LeastSquaresStrategy to compute the coefficients based on
+      least squares.
+    * Uses LeastSquaresMetaModelSelectionFactory to use the LARS selection method.
+    * Uses FixedStrategy in order to keep all the coefficients that the
+      LARS method selected.
+
+    Parameters
+    ----------
+    inputTrain : ot.Sample
+        The input design of experiments.
+    outputTrain : ot.Sample
+        The output design of experiments.
+    multivariateBasis : ot.Basis
+        The multivariate chaos basis.
+    basisSize : int
+        The size of the function basis.
+    distribution : ot.Distribution.
+        The distribution of the input variable.
+    sparse: bool
+        If True, create a sparse PCE.
+
+    Returns
+    -------
+    result : ot.PolynomialChaosResult
+        The estimated polynomial chaos.
+    """
+    if sparse:
+        selectionAlgorithm = ot.LeastSquaresMetaModelSelectionFactory()
+    else:
+        selectionAlgorithm = ot.PenalizedLeastSquaresAlgorithmFactory()
+    projectionStrategy = ot.LeastSquaresStrategy(
+        inputTrain, outputTrain, selectionAlgorithm
+    )
+    adaptiveStrategy = ot.FixedStrategy(multivariateBasis, basisSize)
+    chaosAlgorithm = ot.FunctionalChaosAlgorithm(
+        inputTrain, outputTrain, distribution, adaptiveStrategy, projectionStrategy
+    )
+    chaosAlgorithm.run()
+    chaosResult = chaosAlgorithm.getResult()
+    return chaosResult
+
+
+# %%
+# The next function plots an output sample Y depending
+# on an input sample X.
+
+
+# %%
+def plotXvsY(sampleX, sampleY):
+    """
+    Plot a Y sample against a X sample on a grid.
+
+    Parameters
+    ----------
+    sampleX : ot.Sample(sampleSize, inputDimension)
+        The input sample.
+    sampleY : ot.Sample(sampleSize, outputDimension)
+        The output sample.
+
+    Returns
+    -------
+    grid: ot.GridLayout(outputDimension, inputDimension)
+        The grid of plots of all projections of Y vs X.
+    """
+    dimX = sampleX.getDimension()
+    dimY = sampleY.getDimension()
+    descriptionX = sampleX.getDescription()
+    descriptionY = sampleY.getDescription()
+    grid = ot.GridLayout(dimY, dimX)
+    for i in range(dimY):
+        for j in range(dimX):
+            graph = ot.Graph("", descriptionX[j], descriptionY[i], True, "")
+            cloud = ot.Cloud(sampleX[:, j], sampleY[:, i])
+            graph.add(cloud)
+            if j == 0:
+                graph.setYTitle(descriptionY[i])
+            else:
+                graph.setYTitle("")
```

Comment:
maybe graph.setYTitle(descriptionY[i] if j==0 else "")

Reply:
I removed this code, and replaced it with the fresh ot.VisualTest.DrawPairsXY() available now that #2753 has been merged. This is fixed now.